src/monitor/monitor.c

```diff
@@ -223,19 +226,102 @@ struct mon_init_conn {
 
 static int add_svc_conn_spy(struct mt_svc *svc);
 
+static int service_not_found(char *svc_name,
+                             struct mt_svc **_svc,
+                             bool *_explicitly_configured)
+{
+    DEBUG(SSSDBG_FATAL_FAILURE,
+         "Unable to find peer [%s] in list of services,"
+         " killing connection!\n", svc_name);
+
+    *_svc = NULL;
+    *_explicitly_configured = true;
+    return ENOENT;
+}
+
+#ifdef HAVE_SYSTEMD
+static int socket_activated_service_not_found(struct mon_init_conn *mini,
+                                              char *svc_name,
+                                              bool is_provider,
+                                              struct mt_svc **_svc,
+                                              bool *_explicitly_configured)
+{
+    struct mt_svc *svc = NULL;
+    int ret;
+
+    if (is_provider) {
+        return service_not_found(svc_name, _svc, _explicitly_configured);
+    }
+
+    /* As the service is a responder and wasn't part of the services' list, it means
+     * the service has been socket/dbus activated and has to be configured and added
+     * to the services' list now */
+
+    *_explicitly_configured = false;
+    *_svc = NULL;
+
+    if (check_service(svc_name) != NULL) {
+        DEBUG(SSSDBG_FATAL_FAILURE, "Invalid service %s\n", svc_name);
+        return EINVAL;
+    }
+
+    mini->ctx->services_started = true;
+    mini->ctx->num_services++;
```

Comment:
When shutting the service down: https://github.com/fidencio/sssd/commit/fb6eb85b7032788ca069569e8506816280d03169#diff-97c5bf1b61bdcd18b6593064753142c1R2484